frontend/src/components/custom-tools/combined-output/CombinedOutput.jsx

```diff
@@ -94,53 +117,42 @@ function CombinedOutput({ docId, setFilledFields }) {
             }
           }
           setCombinedOutput(output);
-          return;
-        }
-        const output = {};
-        prompts.forEach((item) => {
-          if (item?.prompt_type === promptType.notes) {
-            return;
-          }
-          output[item?.prompt_key] = "";
-
-          const profileManager = selectedProfile || item?.profile_manager;
-          const outputDetails = data.find(
-            (outputValue) =>
-              outputValue?.prompt_id === item?.prompt_id &&
-              outputValue?.profile_manager === profileManager
-          );
-
-          if (!outputDetails) {
-            return;
-          }
-
-          output[item?.prompt_key] = displayPromptResult(
-            outputDetails?.output,
-            false
-          );
-
-          if (outputDetails?.output?.length > 0) {
-            filledFields++;
-          }
-        });
-
-        setCombinedOutput(output);
-
-        if (setFilledFields) {
-          setFilledFields(filledFields);
+        } else {
+          const output = {};
+          prompts.forEach((item) => {
```

Comment:
NIT: If the goal is to transform an array in this case `prompts` to a single obj `output`, it might be easier to do it using reduce here to improve readability, assuming we are not doing anything else in the loop. 
Just a thought.
@tahierhussain

Reply:
@athul-rs Thanks for the suggestion. I have made the necessary changes.